themes/startwords/layouts/article/single.txt

```diff
@@ -28,6 +28,8 @@
 |  {{ .Parent.Date.Format "January 2006" }}
 |
 +----------------------------------------------------------------------------------->
+{{/* remove all tags; convert footnote list; convert markdown URLs and images; remove heading marker.
+    (could we configure in site config or data file and apply iteratively?)*/}}
+
+{{ .RawContent | replaceRE "!\\[([^\\]]+)\\]\\([^\\)]+\\)" "[image: $1]" | replaceRE "{{\\<[^\\>]+>}}" "" | replaceRE "(?m)^#+ " "" | replaceRE "(?m)(<\\/?[^>]+>)" "" | replaceRE "(?m)^\\[\\^(\\d+)\\]: " "$1. " | replaceRE "\\[([^\\]]+)\\]\\(([^\\)]+)\\)" "$1 [URL: $2]" | replaceRE "\\*\\*([^*]+)\\*\\*" "$1"  }}
```

Comment:
@gwijthoff here's where the regexes are currently! in the single article text template

Reply:
Are we sure this isn't ASCII art? Always amazes me when first looking at them that regexes are actually functional! 😀

I don't have enough knowledge to do much with this, but can always ask @rlskoeser if I notice any issues during the editing process. For now, the transformations on the other side look brilliant, and I'm so excited to have such a beautiful and functional .TXT file.

The one thing I can think of is using a `====` or `-----` underline for section headings, rather than `#` or `##`, respectively. Both pandoc and github-flavored markdown accept that underlining format for section headings, and they also have the benefit of being pretty striking to a reader's eyes on the page.

So, is there a way to edit the above regex in this `single.txt` file to have a section heading display as, for example:

```
Weaving Derrida's References
============================
```

I suppose the regex would have to count the number of characters in the header text and then print the same number of `=`'s. 28 characters in that header text above, and 28 `=`'s beneath it.

Reply:
The regexes are a lot less readable than usual because I had to use double slashes to escape the slashes (took me a bit to figure out). I figured out at some point that I could use ` instead of " in which case I didn't seem to need the doubles, but I was worried about changing it and breaking working regexes!

I like your idea about the alternate header format! It looks like common mark / goldmark supports this syntax too — if we just use that format in the original markdown then we don't have to transform it at all for the plain text version. I can test this.